lib/ansible/galaxy/role.py

```diff
@@ -71,17 +71,11 @@ def __init__(self, galaxy, name, src=None, version=None, scm=None, path=None):
                 path = os.path.join(path, self.name)
             self.path = path
         else:
-            for role_path_dir in galaxy.roles_paths:
-                role_path = os.path.join(role_path_dir, self.name)
-                if os.path.exists(role_path):
-                    self.path = role_path
-                    break
-            else:
-                # use the first path by default
-                self.path = os.path.join(galaxy.roles_paths[0], self.name)
-                # create list of possible paths
-                self.paths = [x for x in galaxy.roles_paths]
-                self.paths = [os.path.join(x, self.name) for x in self.paths]
+            # use the first path by default
+            self.path = os.path.join(galaxy.roles_paths[0], self.name)
+            # create list of possible paths
+            self.paths = [x for x in galaxy.roles_paths]
```

Comment:
In order to skip the first path which is used by default, should not that be `[x for x in galaxy.roles_paths[1:]]` ?

Reply:
@pilou- the "recovery" logic for the `try`/`except` in the install method that handles the "first path found can not be written to, try the next one" depends on the path existing in `self.paths` because it checks the index and steps through the list. https://github.com/ansible/ansible/blob/42a0d7141339295cf7fd2f5ed05d7ec2ef5cb8c1/lib/ansible/galaxy/role.py#L337